piet-coregraphics/src/text.rs

```diff
@@ -102,6 +107,113 @@ struct Attributes {
     style: Option<Span<FontStyle>>,
 }
 
+#[derive(Clone)]
+struct CoreTextFontKey {
+    font: FontFamily,
+    weight: FontWeight,
+    italic: bool,
+    size: f64,
+}
+
+impl PartialEq for CoreTextFontKey {
+    fn eq(&self, other: &CoreTextFontKey) -> bool {
+        self.font == other.font
+            && self.weight == other.weight
+            && self.italic == other.italic
+            && self.size.to_bits() == other.size.to_bits()
+    }
+}
+
+impl Eq for CoreTextFontKey {}
+
+impl Hash for CoreTextFontKey {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.font.hash(state);
+        self.weight.hash(state);
+        self.italic.hash(state);
+        self.size.to_bits().hash(state);
+    }
+}
```

Comment:
just taking a last look, is there a reason we can't derive these?

Reply:
It's the `to_bits()` on floats :(